src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,462 @@
+It takes care of the usual order of events as they would occur in actual user interactions.
+For example, this means that a `click` event from the user would be accompanied by `mousemove`, `hover`, `keydown`, `focus`, `keyup`, and `keypress`.
+
+The most convenient events to test are typically `click`, `keyboard` and `pointer` (to simulate touch events). 
+To dive deeper into these events, you can learn about them in the [`user-event` documentation](https://testing-library.com/docs/user-event/intro).
+
+
+#### Using timers
+
+If you require a fake timer and want to use `vi.useFakeTimers()` in your tests, it must set it up with an `advanceTimers` option:
+
+```tsx title="user-event.test.tsx"
+import { vi } from "vitest"
+
+const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime })
+
+vi.useFakeTimers()
+
+describe("pre-login: sign-in", () => {
+  const { getByRole, getByLabelText } = render(() => <User />)
+  const signUp = getByRole('button', { text: 'Sign-in' })
+  // use convenience API click:
+  user.click(signUp)
+  const name = getByLabelText('Name')
+  // use complex keyboard input:
+  user.keyboard(name, "{Shift}test{Space}{Shift}user")
+  const password = getByLabelText('Password')
+  user.keyboard(name, "secret")
+  const login = getByRole('button', { text: 'Login' })
+  // use touch event
+  user.pointer([
+    { keys: "[TouchA]" target: login },
+    { keys: "[/TouchA]", target: login }
+  ])
+});
+```
+
+### Validating assertions
+
+`vitest` comes with the `expect` function to facilitate assertions that works like:
+
+```tsx frame="none"
+expect(subject)[assertion](value)
+```
+
+The command supports assertions like `toBe` (reference comparison) and `toEqual` (value comparison) out of the box.
+For testing inside the DOM, the package `@testing-library/jest-dom` augments it with some helpful additional assertions:
+
+- [`.toBeInTheDocument()`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobeinthedocument) - checks if the element actually exists in the DOM
+- [`.toBeVisible()`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobevisible) - checks if there is no reason the element should be hidden
+- [`.toHaveTextContent(content)`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohavetextcontent) - checks if the text content matches
+- [`.toHaveFocus()`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohavefocus) - checks if this is the currently focused element
+- [`.toHaveAccessibleDescription(description)`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohaveaccessibledescription) - checks accessible description
+- and a [lot more](https://github.com/testing-library/jest-dom?tab=readme-ov-file#custom-matchers).
+
+
+## Directive testing
+
+[Directives](/reference/jsx-attributes/use) are reusable behaviors for elements.
+They get a reference to the HTML element and the property from the directive prop as an accessor as arguments.
```

Comment:
This sentence is throwing me off a bit - does this rewording make sense? The last part of the sentence (`as an accessor as arguments`) is the part that is confusing.

```suggestion
They receive an accessor as an argument for the HTML element and the property from the direct prop.
```

Reply:
Again, this is technically wrong. The type signature of a directive looks like this: `(ref: HTMLElement, prop: Accessor<any>) => void`. I've tried to make it even more obvious:

"They receive the HTML element they are bound to as their first and an accessor of the directive prop as their second argument."